core/range/minmax_spec.rb

```diff
@@ -0,0 +1,168 @@
+require_relative '../../spec_helper'
+
+describe 'Range#minmax' do
+  before(:each) do
+    @x = mock('x')
+    @y = mock('y')
+
+    @x.should_receive(:<=>).with(@y).any_number_of_times.and_return(-1) # x < y
+    @x.should_receive(:<=>).with(@x).any_number_of_times.and_return(0) # x == x
+    @y.should_receive(:<=>).with(@x).any_number_of_times.and_return(1) # y > x
+    @y.should_receive(:<=>).with(@y).any_number_of_times.and_return(0) # y == y
+  end
+
+  describe 'on an inclusive range' do
+    ruby_version_is '2.6'...'2.7' do
+      # Endless ranges introduced in 2.6
+      it 'should try to iterate endlessly on an endless range' do
+        @x.should_receive(:succ).once.and_return(@y)
+
+        # Endless range literal would cause SyntaxError prior to 2.6
+        range = Range.new(@x, nil)
```

Comment:
`eval("(@x..)")` can be used to avoid the SyntaxError and still use and endless Range (as in other specs).
Can you use that? (and similar in other specs)